src/library/exporter.py

```diff
@@ -52,3 +52,59 @@ def export_quest_to_library(*, source_schema, quest_import_id):
         imported_quest.save()
 
     return result
+
+
+def export_campaign_to_library(*, source_schema, campaign_import_id):
+    """
+    Exports a campaign (Category) and all its quests from the given tenant schema
+    into the shared library schema.
+
+    The exported quests and campaign will not be published automatically in the library.
+
+    Args:
+        source_schema (str): The tenant schema where the campaign currently resides.
+        campaign_import_id (UUID): The import ID of the campaign to export.
+
+    Returns:
+        ImportResult: The result of the import operation into the library schema.
+
+    Raises:
+        Category.DoesNotExist: If the campaign is not found in the source schema.
+        ValidationError: If no published quests exist for the campaign.
+    """
+
+    with schema_context(source_schema):
+        category = Category.objects.get(import_id=campaign_import_id)
+        quests = Quest.objects.filter(published=True, campaign=category)
+        if not quests.exists():
+            raise ValidationError("Cannot export a campaign without any published quests.")
+        export_data = QuestResource().export(quests)
+
+    with library_schema_context():
+        try:
+            result = QuestResource().import_data(
+                export_data,
+                dry_run=False,
+                raise_errors=True,
+                use_transactions=True,
+                import_campaign=True,
+            )
+        except (ValidationError, IntegrityError) as e:
+            # Raise a clearer validation error with context
+            raise ValidationError(f"Failed to import campaign to library schema: {e}") from e
+
+        # Unpublish exported quests
+        exported_quests = Quest.objects.filter(import_id__in=[q.import_id for q in quests])
+        for eq in exported_quests:
```

Comment:
Code comment with context for why this loop is needed.   Will staff get a notification for EVERY quest in the campaign that is sent to the library?  Maybe we don't need signals to trigger on this on the library?  Need some more comments why we are doing it this way and what the consequences are.

Reply:
I'm curious how much lag there is when doing this for real, since it's not asynchronous, how slow is this process?  If I export a campaign of 20 quests, does it take 2 seconds or 10 or 30?

Reply:
Anything that takes more than a few seconds should be done asychronously as a celery task.  I don't think you've done any of these yet, so might as well learn it!

You'll need to update the message so that it lets the user know the quests are being pushed to the library in the background.

Create a celery tasks and run it.  The user will have their next page load right away and the actually processing will be done concurrently.

Reply:
Alternately, you an just leave the message as is, is there any difference to the user whether the quests appear instantly in the library or take 10 seconds?

Reply:
An example if you want one, is the "regenerate all maps" view `regenerate_all` in djcytoscape/views.py.

It checks if there are more than 5 maps to regenerate, it sends it as a celery task/parallel instead of doing it in series.

Reply:
One more thing, is that local development environment might be a lot slower than production, so maybe test it in production first.  If uploading 30 quests takes 4 seconds or less in production, we can leave it as is.